lib/Rex/Fork/Task.pm

```diff
@@ -56,7 +56,18 @@ sub start {
       exit();
     }
     else {
-      &$func($self);
+      eval { &$func($self) };
+      my $exit_code = $@ ? ($? || 1) : 0;
+      push @PROCESS_LIST, $exit_code;
```

Comment:
As a general rule I agree with exiting with a non-OK error code unless everything is OK. Do you think we can somehow add a test to check for this behaviour?